app/src/main/java/com/yterletskyi/happyfriend/features/pin/ui/PinViewModel.kt

```diff
@@ -42,17 +43,30 @@ class PinViewModel @Inject constructor(
     }
 
     private fun authorize() {
-        if (pinCodeController.getPinCode() == null) {
-            pinCodeController.savePinCode(pin)
-            directionsData.value = PinFragmentDirections.toPinScreen(R.string.pin_repeat_title)
+        if (tempPin.value == "") {
```

Comment:
instead of empty string let's better make the pin nullable and check if it's `null`. Null means that there is no value - which logically suits our case better than empty value.

Reply:
I set null to temp pin when user has created new pin code, but while user hasn't pin i must to set empty line.